src/main/dialog.js

```diff
@@ -0,0 +1,163 @@
+ */
+"use strict";
+
+var fluid         = require("infusion");
+var BrowserWindow = require("electron").BrowserWindow;
+
+var gpii  = fluid.registerNamespace("gpii");
+
+require("./utils.js");
+
+
+/**
+ * Base dialog component that supply initialization of
+ * an Electron `BrowserWindow` and the generation of
+ * the file URL that it to be loaded in the same `BrowserWindow`.
+ * NOTE: The generated URL is always relative to the working
+ * directory of the application (`module.terms()`)
+ *
+ * It also Supplies simple interface for show/hide operations of
+ * the window and handles Electron objects cleanup upon destruction.
+ *
+ * Requires:
+ * - (optional) `attrs` - used as raw options for `BrowserWindow` generation.
+ *   For full options list:  https://github.com/electron/electron/blob/master/docs/api/browser-window.md
+ * - relative path from the application's working directory
+ *    - `fileSuffixPath` - the suffix to the file
+ *    - `filePrefixPath` (optional) - the prefix to the file
+ *
+ *   For example a relative path such as `"/src/rendered/waitDialog/index.html"`,
+ *   might be split to:
+ *   `prefixPath = "src/renderer"`
+ *   `fileSuffixPath = "waitDialog/index.html"`
+ */
+fluid.defaults("gpii.app.dialog", {
+    gradeNames: ["fluid.component"],
+
+    config: {
+        attrs: {        // raw attributes used in `BrowserWindow` generation
+            width: 800,
+            height: 600,
+            show: false,
+            frame: false,
+            transparent: true,
+            alwaysOnTop: true,
+            skipTaskbar: true
+        },
+        filePrefixPath: "src/renderer",
+        fileSuffixPath: null,           // e.g. "waitDialog/index.html"
+        url: {
+            expander: {
+                funcName: "gpii.app.dialog.buildFileUrl",
+                args: [
+                    "{that}.options.config.filePrefixPath",
+                    "{that}.options.config.fileSuffixPath"
+                ]
+            }
+        }
+    },
+    members: {
+        dialog: {
+            expander: {
+                funcName: "gpii.app.dialog.makeDialog",
+                args: [
+                    "{that}.options.config.attrs",
+                    "@expand:{that}.getWindowPosition()",
+                    "{that}.options.config.url"
+                ]
+            }
+        }
+    },
+    listeners: {
+        "onDestroy.cleanupElectron": {
+            this: "{that}.dialog",
+            method: "destroy"
+        }
+    },
+    invokers: {
+        getWindowPosition: {
+            funcName: "gpii.app.getWindowPosition",
+            args: [
+                "{that}.options.config.attrs.width",
+                "{that}.options.config.attrs.height"
+            ]
+        },
+        resetWindowPosition: {
+            funcName: "gpii.app.setWindowPosition",
+            args: ["{that}.dialog", "@expand:{that}.getWindowPosition()"]
+        },
+        // Simple default behaviour
+        show: {
```

Comment:
Turn these into namespaced modelListener to the "isShown" field as described below in restartDialog comments